projects/storefrontlib/src/cms-components/myaccount/my-coupons/my-coupons.component.service.spec.ts

```diff
@@ -51,7 +51,11 @@ describe('MyCouponsComponentService', () => {
         cxRoute: 'search',
         params: { query: RELEVANCE },
       },
-      { couponcode: 'CouponForAllProduct' }
+      {
+        state: {
+          couponcode: 'CouponForAllProduct',
+        },
+      }
```

Comment:
we should use property `queryParams` for query params, but not `state`; similar to https://github.com/SAP/spartacus/pull/12638#discussion_r648196574